content/en/docs/releases/supported-releases/index.md

```diff
@@ -73,19 +73,17 @@ Please keep up-to-date and use a supported version.
 | 1.20.x           | 1.20.0+                                              |
 | 1.19.x           | 1.19.3+                                              |
 | 1.18.x           | 1.18.5+                                              |
-| 1.17.x           | 1.17.8+                                              |
 
 ## Supported Envoy Versions
 
 Istio's data plane is based on [Envoy](https://github.com/envoyproxy/envoy).
 
 The relationship between the two project's versions:
 
-| Istio version | Envoy version |
-| ------------- | ------------- |
-| 1.20.x        | 1.28.x        |
-| 1.19.x        | 1.27.x        |
-| 1.18.x        | 1.26.x        |
-| 1.17.x        | 1.25.x        |
+| Istio version | Envoy release branch |
+| ------------- | -------------------- |
+| 1.20.x        | release/v1.28        |
+| 1.19.x        | release/v1.27        |
+| 1.18.x        | release/v1.26        |
 
-In general, Istio releases tend to map one to one with Envoy releases. You can find the precise Envoy commit used by Istio in [`istio/proxy`](https://github.com/istio/proxy/blob/master/WORKSPACE#L38).
```

Comment:
We don't truly map to Envoy versions, like 1.28.1 since we move to the latest commit in the related branch on a daily basis.